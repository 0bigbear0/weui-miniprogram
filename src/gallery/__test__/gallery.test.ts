import path from 'path'
import simulate from 'miniprogram-simulate'

describe('gallery', () => {
    const gallery = simulate.load(path.resolve(__dirname, '../gallery'))

    const id = simulate.load({
        compiler: 'official',
        rootPath: __dirname,
        template: `
            <mp-gallery show="{{show}}" bindchange="change" binddelete="delete" bindhide="hide" img-urls="{{imgUrls}}" delete hide-on-click="{{true}}" current="1"></mp-gallery>
        `,
        usingComponents: {
            'mp-gallery': gallery
        },
        data: {
            imgUrls: [
                'http://desk-fd.zol-img.com.cn/g5/M00/02/05/ChMkJ1bKyZmIWCwZABEwe5zfvyMAALIQABa1z4AETCT730.jpg',
                'http://desk-fd.zol-img.com.cn/g5/M00/02/05/ChMkJ1bKyZmIWCwZABEwe5zfvyMAALIQABa1z4AETCT730.jpg',
                'http://desk-fd.zol-img.com.cn/g5/M00/02/05/ChMkJ1bKyZmIWCwZABEwe5zfvyMAALIQABa1z4AETCT730.jpg'
            ],
            show: true
        },
        methods: {
            change(e) {
                // console.log('current index has changed', e.detail)
            },
            delete(e) {
                // console.log('delete', e.detail)
            },
            hide() {
                // console.log('component hide')
                // this.setData({
                //     show: true
                // })
                // setTimeout(() => {
                //     // console.log('component show')
                //     this.setData({
                //         show: true
                //     })
                // }, 1000)
            }
        }
    })

    test('basic', async () => {
        const comp = simulate.render(id)
        comp.attach(document.createElement('parent-wrapper'))
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
    })
})
