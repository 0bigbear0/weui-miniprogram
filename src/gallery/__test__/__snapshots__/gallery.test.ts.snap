// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gallery basic 1`] = `
<main
  delete="{{undefined}}"
  delete="{{undefined}}"
  bind:change="change"
  bind:delete="delete"
  bind:hide="hide"
>
  <wx-view
    class="weui-gallery weui-gallery_show "
  >
    <wx-view
      class="weui-gallery__info"
    >
      11/3
    </wx-view>
    <wx-swiper
      autoplay="{{false}}"
      class="weui-gallery__img__wrp"
      current="1"
      duration="{{500}}"
      indicatorDots="{{false}}"
      bind:change="change"
      bind:tap="hideGallery"
    >
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://desk-fd.zol-img.com.cn/g5/M00/02/05/ChMkJ1bKyZmIWCwZABEwe5zfvyMAALIQABa1z4AETCT730.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://desk-fd.zol-img.com.cn/g5/M00/02/05/ChMkJ1bKyZmIWCwZABEwe5zfvyMAALIQABa1z4AETCT730.jpg"
        />
      </swiper-item>
      <swiper-item>
        <wx-image
          class="weui-gallery__img"
          mode="aspectFit"
          src="http://desk-fd.zol-img.com.cn/g5/M00/02/05/ChMkJ1bKyZmIWCwZABEwe5zfvyMAALIQABa1z4AETCT730.jpg"
        />
      </swiper-item>
    </wx-swiper>
    <wx-view
      class="weui-gallery__opr"
    >
      <wx-navigator
        class="weui-gallery__del"
        bind:tap="deleteImg"
      >
        删除
      </wx-navigator>
    </wx-view>
  </wx-view>
</main>
`;
